### PR TITLE
doxygen config generation

### PR DESCRIPTION
**Pull request by grevaillot**
Hi,

This PR suggests a way to get rid of plenty of copypasted doxygen stuff, by generating doc root doxygenlayout file, as well as devices stuff, based on main Makefile $(TARGETS) and template files.

This also allows to build only one device doc easily (ie: make TARGETS=stm32f0 doc), making documentation fixing less painfull, and make doc folder device agnostic.

We currently loose device "fancy" naming as device name is guessed (toUpper()..) from folder name - so currently, devices like efm32hg are not known as Happy Gecko anymore, but I currently dont see any elegant way to do that while keeping the doc folder device agnostic. Maybe we could define that in device's makefile ? I'm open to suggestions..

### CODE REVIEW

File: scripts/gendoxylayout.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+from lxml import etree
```
Comment (karlp) [inline scripts/gendoxylayout.py]: lxml is a non-core library right?  and it's one that has to be compiled?

Comment (karlp) [reply scripts/gendoxylayout.py]: if I'm reading right, the expat binding is the only one standard.  https://docs.python.org/2/library/xml.html (and same for v3)  If we use that, it's less deps to install.  I like the idea of this PR, but I don't really like adding deps for this

Comment (grevaillot) [reply scripts/gendoxylayout.py]: arf, It looked python standard here - I'll have a look at that.

**Comment by grevaillot** [comment]
Ok, I just pushed a modified tool using xml.etree.ElementTree - apparently ok on python2/3 out of the box.
ElementTree stuff is not as convenient to use as lxml, so it's a bit less nice ..

**Comment by karlp** [comment]
excellent thanks.  tested with both python2.7 and 3. something.  used it to turn ondocs for the gd32 nice and quickly too.

**Comment by karlp** [comment]
(merrged in bdc38eed1 and the one before)

**Comment by grevaillot** [comment]
nice, cool :)

**Pull request closed without merging**